packages/peregrine/lib/talons/MagentoRoute/useMagentoRoute.js

```diff
@@ -13,6 +14,13 @@ export const useMagentoRoute = (props = {}) => {
     const { replace } = useHistory();
     const { pathname } = useLocation();
     const [componentMap, setComponentMap] = useRootComponents();
+    const [previousPathname, setPreviousPathname] = useState(null);
```

Comment:
Yes, this is probably the right place for this logic.

Reply:
Nice!
There's actually a bug in this branch when navigating to routes not covered by useMagentoRoute (it holds on to the previous component). The fix for this is in [PWA-1204](https://github.com/magento/pwa-studio/pull/3353/files#diff-954aac140c96fefc0d3084918ee698a35f69001c61f320bfd641a4c3cc804d05R164), by setting it back to null when unmounting